pkg/providers/amifamily/ami.go

```diff
@@ -133,6 +135,17 @@ func (p *DefaultProvider) DescribeImageQueries(ctx context.Context, nodeClass *v
 	return queries, nil
 }
 
+func (p *DefaultProvider) updateFilterFromCustomParameter(ctx context.Context, ssmParameterName string, idFilter *ec2types.Filter) {
+	imageID, err := p.ssmProvider.GetCustomParameter(ctx, ssm.Parameter{
+		Name: ssmParameterName,
+	})
+	if err != nil {
+		log.FromContext(ctx).WithValues("ssmParameterName", ssmParameterName).V(1).Error(err, "parameter not found")
+	} else {
+		idFilter.Values = append(idFilter.Values, imageID)
+	}
```

Comment:
style: we prefer early returns for errors than if-else in this project, it's less error prone when adding additional checks. For example:

```go
	if err != nil {
		log.FromContext(ctx).WithValues("ssmParameterName", ssmParameterName).V(1).Error(err, "parameter not found")
        return
	} 
	idFilter.Values = append(idFilter.Values, imageID)
```